src/cli/pm_pkg_command.zig

```diff
@@ -0,0 +1,784 @@
+const logger = bun.logger;
+const js_ast = bun.js_ast;
+const JSPrinter = bun.js_printer;
+const JSON = bun.JSON;
+
+pub const PmPkgCommand = struct {
+    const SubCommand = enum {
+        get,
+        set,
+        delete,
+        fix,
+        help,
+
+        fn fromString(str: []const u8) ?SubCommand {
+            return std.meta.stringToEnum(SubCommand, str);
+        }
+    };
+
+    pub fn exec(ctx: Command.Context, pm: *PackageManager, positionals: []const string, cwd: []const u8) !void {
+        if (positionals.len <= 1) {
+            printHelp();
+            return;
+        }
+
+        const subcommand = SubCommand.fromString(positionals[1]) orelse {
+            Output.errGeneric("Unknown subcommand: {s}", .{positionals[1]});
+            printHelp();
+            Global.exit(1);
+        };
+
+        switch (subcommand) {
+            .get => try execGet(ctx, pm, positionals[2..], cwd),
+            .set => try execSet(ctx, pm, positionals[2..], cwd),
+            .delete => try execDelete(ctx, pm, positionals[2..], cwd),
+            .fix => try execFix(ctx, pm, cwd),
+            .help => printHelp(),
+        }
+    }
+
+    fn printHelp() void {
+        Output.prettyln("<r><b>bun pm pkg<r> <d>v" ++ Global.package_json_version_with_sha ++ "<r>", .{});
+        const help_text =
+            \\  Manage data in package.json
+            \\
+            \\<b>Subcommands<r>:
+            \\  <cyan>get<r> <blue>[key ...]<r>          Get values from package.json
+            \\  <cyan>set<r> <blue>key=value ...<r>      Set values in package.json
+            \\    <d>└<r> <cyan>--json<r>             Parse values as JSON (e.g. {{"a":1}})
+            \\  <cyan>delete<r> <blue>key ...<r>         Delete keys from package.json
+            \\  <cyan>fix<r>                    Auto-correct common package.json errors
+            \\
+            \\<b>Examples<r>:
+            \\  <d>$<r> <b><green>bun pm pkg<r> <cyan>get<r> <blue>name version<r>
+            \\  <d>$<r> <b><green>bun pm pkg<r> <cyan>set<r> <blue>description="My awesome package"<r>
+            \\  <d>$<r> <b><green>bun pm pkg<r> <cyan>set<r> <blue>keywords='["test","demo","example"]'<r> <cyan>--json<r>
+            \\  <d>$<r> <b><green>bun pm pkg<r> <cyan>set<r> <blue>config='{{"port":3000,"debug":true}}'<r> <cyan>--json<r>
+            \\  <d>$<r> <b><green>bun pm pkg<r> <cyan>set<r> <blue>scripts.test="bun test"<r>
+            \\  <d>$<r> <b><green>bun pm pkg<r> <cyan>set<r> <blue>bin.mycli=cli.js<r>
+            \\  <d>$<r> <b><green>bun pm pkg<r> <cyan>delete<r> <blue>scripts.test devDependencies.webpack<r>
+            \\  <d>$<r> <b><green>bun pm pkg<r> <cyan>fix<r>
+            \\
+            \\<b>More info<r>: <magenta>https://bun.com/docs/cli/pm#pkg<r>
+            \\
+        ;
+        Output.pretty(help_text, .{});
+        Output.flush();
+    }
+
+    fn findPackageJson(allocator: std.mem.Allocator, cwd: []const u8) ![]const u8 {
```

Comment:
is the purpose of this function to find the root package.json, or the package.json to be edited/read? I think both are already provided by the PackageManager struct after initialization. I think the one you want is `manager.original_package_json_path`

Reply:
The most local one, for pm version i had the same thing because the only data accessible was actual root which we didn't need here.

I should make it a shared function or something though.